trees/FairTree.py

```diff
@@ -21,225 +29,130 @@ class FairTreeEstimator(BaseEstimator):
+    def fit(self, X, y, P, L):
+        """A reference implementation of a fitting function.
 
         Parameters
         ----------
-        X : array-like, shape (n_samples, n_features)
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
             The training input samples.
-        y : array-like, shape (n_samples,)
-            The target values. An array of int.
+        y : array-like, shape (n_samples,) or (n_samples, n_outputs)
+            The target values (class labels in classification, real numbers in
+            regression).
+        P : array-like, shape (n_samples,) or (n_samples, n_p)
+            The protected feature columns (Race, gender, etc)
+
+        L : array-like, shape (n_samples,) or (n_samples, n_l)
+            The legitimate factor columns (e.g., prior number of criminal acts)
 
         Returns
         -------
         self : object
             Returns self.
         """
-        # Check that X and y have correct shape
+        # store column information and dtypes if any
+        self.extract_metadata(X, y, P, L)
+        # this function returns converted X and y but we retain metadata
         X, y = check_X_y(X, y)
+        # Raises ValueError if there is a column that has values other than 0 or 1
+        check_binary(X)
         # Store the classes seen during fit
         self.classes_ = unique_labels(y)
 
-        self.X_ = X
-        self.y_ = y
-        # Return the classifier
-        return self
-
-    def predict(self, X):
-        """ A reference implementation of a prediction for a classifier.
-
-        Parameters
-        ----------
-        X : array-like, shape (n_samples, n_features)
-            The input samples.
-
-        Returns
-        -------
-        y : ndarray, shape (n_samples,)
-            The label for each sample is the label of the closest sample
-            seen during fit.
-        """
-        # Check is fit had been called
-        check_is_fitted(self, ['X_', 'y_'])
-
-        # Input validation
-        X = check_array(X)
 
-        closest = np.argmin(euclidean_distances(X, self.X_), axis=1)
-        return self.y_[closest]
-
-
-class FairTreeTransformer(TransformerMixin, BaseEstimator):
-    """ An example transformer that returns the element-wise square root.
-
-    For more information regarding how to build your own transformer, read more
-    in the :ref:`User Guide <user_guide>`.
-
-    Parameters
-    ----------
-    demo_param : str, default='demo'
-        A parameter used for demonstation of how to pass and store paramters.
-
-    Attributes
-    ----------
-    n_features_ : int
-        The number of features of the data passed to :meth:`fit`.
-    """
 
-    def __init__(self, demo_param='demo'):
-        self.demo_param = demo_param
+        'Here we need to convert P and L to np.arrays. We need a function.' \
+        'We also need to check that P.shape and L.shape is not (n_sample, )'
```

Comment:
Change to comment. So we need a function to convert to np.arrays and check the shape of the arrays? What shape should these arrays be?